src/Kentico.Xperience.Siteimprove/Services/SiteimproveService.cs

```diff
@@ -0,0 +1,274 @@
+
+
+        private string GetDomain()
+        {
+            return RequestContext.CurrentDomain;
+        }
+
+
+        private async Task<bool> TryEnableContentCheck(HttpClient client, CancellationToken? cancellationToken = null)
+        {
+            var response = await Post(SiteimproveConstants.CONTENT_CHECK_PATH, cancellationToken, client);
+
+            if (response == null)
+            {
+                return false;
+            }
+
+            if (!response.IsSuccessStatusCode)
+            {
+                string message = response.StatusCode == HttpStatusCode.PaymentRequired
+                    ? "Cannot enable content check. User not subscribed to Prepublish feature."
+                    : "An error occurred during enabling of content check.";
+
+                eventLogService.LogError(nameof(SiteimproveService), nameof(TryEnableContentCheck), message);
+
+                return false;
+            }
+
+            return true;
+        }
+
+
+        private async Task<bool> IsContentCheckReady(HttpClient client, CancellationToken? cancellationToken = null)
+        {
+            var contentCheck = await Get<SiteimproveContentCheck>(SiteimproveConstants.CONTENT_CHECK_PATH, cancellationToken, client);
+
+            if (contentCheck == null)
+            {
+                return false;
+            }
+
+            return contentCheck.IsReady;
+        }
+
+
+        private async Task<string> GetTokenInternal(CancellationToken? cancellationToken = null)
+        {
+            string token = SettingsKeyInfoProvider.GetValue(SiteimproveConstants.TOKEN_SETTINGS_KEY_NAME);
+
+            if (string.IsNullOrEmpty(token))
+            {
+                string requestPath = string.Format(SiteimproveConstants.TOKEN_URL, Uri.EscapeDataString(SiteimproveConstants.CMS_NAME));
+                var receivedToken = await Get<SiteimproveToken>(requestPath, cancellationToken);
+
+                if (receivedToken == null || receivedToken.Value == null)
+                {
+                    return token;
+                }
+
+                token = receivedToken.Value;
+                SetTokenSettings(token);
+            }
+
+            return token;
+        }
+
+
+        private async Task<T?> Get<T>(string requestPath, CancellationToken? cancellationToken = null, HttpClient? client = null) where T : class
+        {
+            client ??= HttpClient;
+
+            try
+            {
+                return await client.GetFromJsonAsync<T>(requestPath, cancellationToken: cancellationToken ?? CancellationToken.None);
+            }
+            catch (Exception ex)
+            {
+                eventLogService.LogException(nameof(SiteimproveService), nameof(Get), ex);
+            }
+
+            return null;
+        }
+
+
+        private async Task<HttpResponseMessage?> Post(string requestPath, CancellationToken? cancellationToken = null, HttpClient? client = null)
+        {
+            client ??= HttpClient;

```

Comment:
I noticed that this assignment guards against the `client` being null, but these calls all being by passing the same `HttpClient` field to the source method.

Is there a reason you wanted to supply an `HttpClient` as a parameter here rather than using the class field `HttpClient`?

Reply:
Yes, I didn't want to fetch multiple instances of `HttpClient` in methods where multiple requests are required. For example, the `CheckPages` would potentially create dozens of `HttpClient`s otherwise.